### ISSUE TITLE
3.x - Flash element path for prefixes

### ISSUE THREAD

**Issue by TamiasSibiricus** [cakephp/cakephp#4320]
Default Flash elements path is: 
~~~
src/Template/Element/Flash
~~~
I think prefixed views needs separate Flash elements sometimes like layouts. Example:
~~~
src/Template/Admin/Element/Flash
~~~

**Comment by HavokInspiration** [comment]
I can see some value in this for Flash Elements (and Elements in general actually).
I also encouter, in most of my apps, the case where Flash elements are different for the front and for the admin side.
For general Elements, I usually create a ``Admin`` folder in ``Elements``. I wouldn't mind switching for all of it.
I think this would be a good addition. And since Layouts and Views already switch, that would be kind of logical that elements can too.
I never used Cells, so I can't say anything about it though.

**Comment by markstory** [comment]
I still don't think that Cells should auto switch, but the arguments for making elements consistent are reasonable.

**Comment by markstory** [comment]
Closing this as it was merged into 3.0.1
